src/components/controls/buttons/DropdownButton.js

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Button from '../../controls/buttons/Button';
+import RootCloseWrapper from 'react-overlays/lib/RootCloseWrapper';
+import styled, { ThemeProvider } from 'styled-components';
+import defaultTheme from '../../theme/defaultTheme';
+
+const Caret = styled.span`
+  display: inline-block;
+  width: 0;
+  height: 0;
+  margin-left: 5px;
+  vertical-align: middle;
+  border-top: 4px dashed;
+  border-right: 4px solid transparent;
+  border-left: 4px solid transparent;
+`;
+
+const ButtonPanel = styled.div`
+  display: ${props => (props.isOpen ? 'block' : 'none')};
+  margin-top: 3px;
+  position: absolute;
+  border: 1px solid ${props => props.theme.colors.grayLight};
+  background-color: ${props => props.theme.colors.white};
+`;
+
+const ButtonPanelChildrenContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const StyledButtonLink = styled(Button)`
+  padding: 10px 20px;
+  color: black;
+  text-decoration: none;
+  text-align: left;
+  margin-bottom: 0px;
+  &:hover {
+    background-color: ${props => props.theme.colors.grayLighter};
+  }
+`;
+
+class DropdownButton extends React.Component {
+  constructor(props) {
+    super();
+    this.state = {
+      buttonValue: props.buttonValue,
+      isOpen: false
+    };
+    this.handleDropdownItemClick = this.handleDropdownItemClick.bind(this);
+  }
+
+  toggleDropdown = () => {
+    this.setState(previousState => ({ isOpen: !previousState.isOpen }));
+  };
+
+  closeDropdown = () => {
+    if (this.state.isOpen) {
+      this.setState({ isOpen: false });
+    }
+  };
+
+  handleDropdownItemClick(button) {
+    return event => {
+      if (this.props.shouldUpdateButtonValue) {
+        this.setState({ buttonValue: button.buttonValue });
+      }
+
+      this.props.onDropdownItemClick(button);
```

Comment:
At this point the following scenario is completely conjecture.

1. We click a button, `buttonValue` is overwritten
2. `onDropdownItemClick` is called
3. The parent of `DropdownButton` responds to `onDropdownItemClick` by passing in a new `buttonValue` to `DropdownButton`
4. `buttonValue` is still the same as what was set in the previous `setState` call, `DropdownButton` does not respond at all to the new prop value of `buttonValue`

At this point I think the only issue with the current approach is that we can't manage the selected value outside of the component, and I'm not entirely sure if this is really a problem with our current use cases.

Reply:
I think for right what we have is fine. The update of the value was put in here specifically for an use case defined in a downstream design. I would say that 95% of the time this value would be false